### PR TITLE
Add v2 of chapter 7 notebook which uses current haystack

### PR DESCRIPTION
**Pull request by tstadel**
In the most recent version of Haystack, the evaluation of pipelines is much different compared to the version described in the (first edition of the) book. Pipeline now has a method eval(), which runs the pipeline given input data from labels and returns the prediction and the labels in the format of one pandas dataframe per pipeline node.
That's why we created a new second version of chapter 7's notebook which makes use of the new Haystack evaluation mechanism. The results of the evaluation between the first and the second version differ because in the previous notebook version there were two bugs: one within the notebook itself during creating the labels that made some labels disappear and one within haystack's previous evaluation logic that caused handling no_answers (e.g. when there is no answer to a certain question) in a improper way.

This PR:
- adds v2 of chapter 07: This contains updates of the haystack API and several bug fixes
- adds new `install_requirements` option `is_chapter7_v2`
- adds `requirements-chpater7-v2.txt` with `farm-haystack==1.4.0`

Misc:
- starting with Haystack 1.4.0 ElasticsearchRetriever is now called BM25Retriever

### CODE REVIEW

**Comment by lvwerra** [comment]
Looks great to me, thanks for working on this! 🚀 

I let @lewtun do the closer code review as he has mainly worked on this.

**Review by lewtun — CHANGES_REQUESTED** [review]
Thanks a lot @tstadel for revamping this chapter to use the new pipeline API 🔥 !

I've left a few nits, but otherwise this looks great - thank you again!

**Comment by tstadel** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/nlp-with-transformers/notebooks/pull/48/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by tstadel** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/nlp-with-transformers/notebooks/pull/48/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by tstadel** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/nlp-with-transformers/notebooks/pull/48/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by tstadel** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/nlp-with-transformers/notebooks/pull/48/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by tstadel** [comment]
<p>Done.</p>

---

 <a href='https://app.reviewnb.com/nlp-with-transformers/notebooks/pull/48/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by tstadel** [comment]
@lewtun I made the respective changes. I found another bug regarding the logits plot in the `Tokenize text for QA` section. The plot now correctly depicts the labels :-)

**Review by lewtun — APPROVED** [review]
Thanks for iterating with the changes @tstadel and apologies for the slow review! Merging this and we can announce it this week :)
